block_massaction.php

```diff
@@ -92,6 +92,11 @@ public function get_content(): stdClass {
         $this->content->footer = '';
 
         if ($this->page->user_is_editing()) {
+            $blockcontext = context_block::instance($this->instance->id);
+            if (!has_capability('block/massaction:use', $blockcontext)) {
+                $this->content->text = get_string('nopermissions', 'error', get_string('massaction:use', 'block_massaction'));
+                return $this->content;
```

Comment:
Wondering why we need to display anything at all? IMHO we don't need to display content at all if user don't have permissions. But let's leave it up to the maintainer to decide.

Reply:
If a user adds the block without use permission, I think we need to show something for the block.
It's probably better not to allow to add the block if user doesn't have use permission.

I'm leaving this as I updated.
Please let me know which solution is better.

Reply:
Block can be already in a course when a user with caps to see the block, but without caps to do anything with mass actions  accessing a course.

Reply:
I think the best method is to show the "No permission" message.  I could see it being confusing if a user added the block but didn't see the block.  They would probably think that the block wasn't added correctly.  For as rare as this case would probably be I think I can live with it.